sno/repo_files.py

```diff
@@ -0,0 +1,87 @@
+import os
+
+from pathlib import Path
+import shlex
+import subprocess
+
+
+from . import is_windows
+from .exceptions import SubprocessError
+
+
+HEAD = "HEAD"
+COMMIT_EDITMSG = "COMMIT_EDITMSG"
+ORIG_HEAD = "ORIG_HEAD"
+MERGE_HEAD = "MERGE_HEAD"
+MERGE_MSG = "MERGE_MSG"
+MERGE_INDEX = "MERGE_INDEX"
+MERGE_LABELS = "MERGE_LABELS"
+
+
+def repo_file_path(repo, filename):
+    return Path(repo.path) / filename
+
+
+def repo_file_exists(repo, filename):
+    return repo_file_path(repo, filename).exists()
+
+
+def write_repo_file(repo, filename, contents):
+    if not isinstance(contents, str):
+        raise TypeError("File contents must be a string", type(contents))
+    if contents[-1] != "\n":
```

Comment:
Could cause IndexError for empty strings. Use `.endswith()`

Reply:
Done